File: js/src/maplestory/api/dto/character/characterItemEquipmentTotalOptionDto.ts
import { CharacterItemEquipmentTotalOptionBody } from '../../response/character/characterItemEquipmentBody';

/**
 * 캐릭터 장비 최종 옵션 정보
 */
class CharacterItemEquipmentTotalOptionDto {
  /**
   * STR
   */
  str: string;

  /**
   * DEX
   */
  dex: string;

  /**
   * INT
   */
  int: string;

  /**
   * LUK
   */
  luk: string;

  /**
   * 최대 HP
   */
  maxHp: string;

  /**
   * 최대 MP
   */
  maxMp: string;

  /**
   * 공격력
   */
  attackPower: string;

  /**
   * 마력
   */
  magicPower: string;

  /**
   * 방어력
   */
  armor: string;

  /**
   * 이동속도
   */
  speed: string;

  /**
   * 점프력
   */
  jump: string;

  /**
   * 보스 공격 시 데미지 증가(%)
   */
  bossDamage: string;

  /**
   * 몬스터 방어율 무시(%)
   */
  ignoreMonsterArmor: string;

  /**
   * 올스탯(%)
   */
  allStat: string;

  /**
   * 데미지(%)
   */
  damage: string;

  /**
   * 착용 레벨 감소
   */
  equipmentLevelDecrease: number;

  /**
   * 최대 HP(%)
   */
  maxHpRate: string;

  /**
   * 최대 MP(%)
   */
  maxMpRate: string;

  constructor(obj: CharacterItemEquipmentTotalOptionBody) {
    const {
      str,
      dex,
      int,
      luk,
      max_hp,
      max_mp,
      attack_power,
      magic_power,
      armor,
      speed,
      jump,
      boss_damage,
      ignore_monster_armor,
      all_stat,
      damage,
      equipment_level_decrease,
      max_hp_rate,
      max_mp_rate,
    } = obj;

    this.str = str;
    this.dex = dex;
    this.int = int;
    this.luk = luk;
    this.maxHp = max_hp;
    this.maxMp = max_mp;
    this.attackPower = attack_power;
    this.magicPower = magic_power;
    this.armor = armor;
    this.speed = speed;
    this.jump = jump;
    this.bossDamage = boss_damage;
    this.ignoreMonsterArmor = ignore_monster_armor;
    this.allStat = all_stat;
    this.damage = damage;
    this.equipmentLevelDecrease = equipment_level_decrease;
    this.maxHpRate = max_hp_rate;
    this.maxMpRate = max_mp_rate;
  }
}

export { CharacterItemEquipmentTotalOptionDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterItemEquipmentTotalOptionDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * 道具最終選項資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterItemEquipmentTotalOptionDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterItemEquipmentTotalOptionDTO {

    /**
     * 力量 (STR)
     */
    @JsonProperty("str")
    private String strength;

    /**
     * 敏捷 (DEX)
     */
    @JsonProperty("dex")
    private String dexterity;

    /**
     * 敏捷 (DEX)
     */
    @JsonProperty("int")
    private String intelligence;

    /**
     * 幸運 (LUK)
     */
    @JsonProperty("luk")
    private String luck;

    /**
     * 最大體力 (HP)
     */
    @JsonProperty("max_hp")
    private String maxHp;

    /**
     * 最大 MP
     */
    @JsonProperty("max_mp")
    private String maxMp;

    /**
     * 攻擊力
     */
    @JsonProperty("attack_power")
    private String attackPower;

    /**
     * 魔力
     */
    @JsonProperty("magic_power")
    private String magicPower;

    /**
     * 防禦力
     */
    @JsonProperty("armor")
    private String armor;

    /**
     * 移動速度
     */
    @JsonProperty("speed")
    private String speed;

    /**
     * 跳躍力
     */
    @JsonProperty("jump")
    private String jump;

    /**
     * 對首領傷害增加 (%)
     */
    @JsonProperty("boss_damage")
    private String bossDamage;

    /**
     * 無視怪物防禦力 (%)
     */
    @JsonProperty("ignore_monster_armor")
    private String ignoreMonsterArmor;

    /**
     * 全部能力值 (%)
     */
    @JsonProperty("all_stat")
    private String allStat;

    /**
     * 傷害 (%)
     */
    @JsonProperty("damage")
    private String damage;

    /**
     * 裝備等級減少
     */
    @JsonProperty("equipment_level_decrease")
    private Long equipmentLevelDecrease;

    /**
     * 最大體力 (HP) (%)
     */
    @JsonProperty("max_hp_rate")
    private String maxHpRate;

    /**
     * 最大 MP (%)
     */
    @JsonProperty("max_mp_rate")
    private String maxMpRate;
}